Lots of code style cleanup

zromano opened the pull request:
This PR is massive and really should've been several PRs. I can break it into separate ones if wanted, but they will need to all go in order.

This PR would best be reviewed commit by commit if not broken up into separate PRs.

Does the following:

- Update QueryUtil and QueryUtilTest
- Use Standard UTF-8 Charset to avoid an unnecessary try/catch
- Use the Map interface instead of directly using Hashmap
- Optimize imports across the entire repo
- Fix Javadoc errors
- Fix public/private modifiers
  - Interface method definitions don't need a public modifier
  - Enums don't need a private modifier on the constructor
- Clean up Enums
  - Make each option have its own line
  - Remove spacing between option and option values
- Fix Error logging to accept the appropriate number of args
  - We shouldn't log the throwable as a "{}" in the message template, the log method will accept it as a param
- Use capital "L" when declaring longs
- Update date in License agreements to be 2023
- Make tests requiring internet connection IT tests
- Remove redundant type arguments
  - Ex Map<String, String> map = new HashMap<String, String> is redundant.
- Migrate from Junit Assertions to AssertJ assertions
  - Also fix order, so it's always assertThat(<ACTUAL>).is(<EXPECTED>)... the old assertions were inconsistent
  - Old tests would say to fail the test instead of using an assertion
  - Some old tests used custom error messages rather than making the test/assertion self-explanatory
- Simplify building comma separated list in QueryUtil
- Separate string concatenation into separate variables and update invalid test assertions
- Update SDK tests so that they do not have a try-catch format
  - If an exception is thrown, just let the test fail. No need to catch it and fail the test manually

smar-zach-romano committed:
Update QueryUtil and QueryUtilTest

smar-zach-romano committed:
Use Standard UTF-8 Charset to avoid an unnecessary try/catch

smar-zach-romano committed:
Use the Map interface instead of directly using Hashmap

smar-zach-romano committed:
Optimize imports across the entire repo

smar-zach-romano committed:
Fix Javadoc errors

smar-zach-romano committed:
Fix public/private modifiers
- Interface method definitions don't need a public modifier
- Enums don't need a private modifier on the constructor

smar-zach-romano committed:
Clean up Enums
- Make each option have its own line
- Remove spacing between option and option values

smar-zach-romano committed:
Fix Error logging to accept the appropriate number of args
 - You shouldn't log the throwable as a "{}" in the message template, the log method will accept it as a param

smar-zach-romano committed:
Use capital "L" when declaring longs

smar-zach-romano committed:
Update date in License agreements to be 2023

smar-zach-romano committed:
Make tests requiring internet connection IT tests

smar-zach-romano committed:
Remove redundant type arguments
- Ex Map<String, String> map = new HashMap<String, String> is redundant

smar-zach-romano committed:
Migrate from Junit Assertions to AssertJ assertions
- Also fix order, so it's always assertThat(<ACTUAL>).is(<EXPECTED>)... the old assertions were inconsistent
- Old tests would say to fail the test instead of using an assertion
- Some old tests used custom error messages rather than making the test/assertion self-explanatory

zromano commented:
Note: If we do not split this into multiple MRs, we should rebase and merge so that we pull in the commit messages and don't just combine this all to a single commit.

DonoA commented on src/main/java/com/smartsheet/api/CommentAttachmentResources.java:
```diff
@@ -105,5 +105,5 @@ public interface CommentAttachmentResources {
 //     * @throws SmartsheetException if there is any other error during the operation
 //     */
-//    public Attachment attachFileWithMultipartUpload(long sheetId, Comment comment, File file, String contentType) throws FileNotFoundException, SmartsheetException;
+//    Attachment attachFileWithMultipartUpload(long sheetId, Comment comment, File file, String contentType) throws FileNotFoundException, SmartsheetException;
```
Wouldn't want anyone getting any ideas 😆

DonoA commented on src/main/java/com/smartsheet/api/internal/EventResourcesImpl.java:
```diff
@@ -76,7 +82,7 @@ public EventResult listEvents(Object since, String streamPosition, Integer maxCo
         String path = "events";
 
-        HashMap<String, Object> parameters = new HashMap<String, Object>();
+        Map<String, Object> parameters = new HashMap<>();
```
Maybe we should start using `var`? In a follow up PR ofc

zromano replied:
I'm open to it.
Can bring that up and decide as a team

DonoA commented on src/main/java/com/smartsheet/api/internal/http/RequestAndResponseData.java:
```diff
@@ -307,17 +306,13 @@ static String binaryBody(HttpEntitySnapshot entity) {
-        } catch (UnsupportedEncodingException badEncodingOrNotText) {
-            contentAsText = new String(Hex.encodeHex(contentBytes));
-        }
+        contentAsText = new String(contentBytes, StandardCharsets.UTF_8);
```
I don't think we want to change this

zromano replied:
We don't have to worry about the `UnsupportedEncodingException` anymore because we are passing in the enum value and not a String. 

The `UnsupportedEncodingException`  would occur if you passed in something like `UFT_9999` or any other invalid encoding type

DonoA replied:
I see, I thought this would also be thrown if the contents could not be decoded but I just checked the docs and that does not appear to be correct.

DonoA commented on src/main/java/com/smartsheet/api/internal/util/QueryUtil.java:
```diff
@@ -20,34 +20,38 @@
             return "";
         }
         StringBuilder result = new StringBuilder();
         for (Object obj : list) {
+            // is there a better way?
             result.append(',').append(obj.toString());
         }
-        return result.length() == 0 ? "" : result.substring(1);
+        // Remove the extra comma at the beginning
+        return result.substring(1);
```
Yes, `list.stream().map(Object::toString).collect(Collectors.joining(","))`

zromano replied:
thanks, I didn't have wifi when I was doing this and could remember off the top of my head 😆 

Will update

zromano replied:
fixed

smar-zach-romano committed:
Simplify building comma separated list in QueryUtil

apederson94 commented on src/integration-test/java/EventResourcesIT.java:
```diff
@@ -48,31 +47,33 @@ void testListEvents() throws SmartsheetException {
         Date lastHour = new Date(System.currentTimeMillis() - TimeUnit.HOURS.toMillis(1));
         EventResult eventResult = smartsheet.eventResources().listEvents(lastHour, null, 10, false);
-        assertTrue(eventResult.getData().size() <= 10);
+        assertThat(eventResult.getData()).hasSizeGreaterThanOrEqualTo(10);
```
This looks like you accidentally inverted the logic here

zromano replied:
Good catch... that would've been hard to find since that test is disabled 😆

apederson94 commented on src/integration-test/java/FavoriteResourcesIT.java:
```diff
@@ -68,7 +73,7 @@ void testRemoveFavorites() throws Exception {
         smartsheet.favoriteResources().addFavorites(new Favorite.AddFavoriteBuilder().addFavorite(folder1.getId(), FavoriteType.FOLDER).addFavorite(folder2.getId(), FavoriteType.FOLDER).build());
 
         smartsheet.favoriteResources().removeFavorites(FavoriteType.FOLDER, new HashSet(Arrays.asList(folder1.getId())));
-        //assertNotNull(smartsheet.favoriteResources().listFavorites(null));
+        //assertThat(smartsheet.favoriteResources().listFavorites(null)).isNotNull();
```
Seems like this could just be deleted if we're just going to leave it commented out

zromano replied:
I agree, but will approach that topic in a subsequent MR.... I'd like to take a look at the logic prior to deleting things

apederson94 commented on src/integration-test/java/SheetResourcesIT.java:
```diff
@@ -179,28 +205,25 @@ public void testCreateSheetInFolder() throws SmartsheetException, IOException {
-                || newSheetFromTemplate.getPermalink().toString().isEmpty()) {
-            fail("Sheet not correctly copied");
-        }
+        assertThat(newSheetFromTemplate.getId().toString()).isEmpty();
```
Should be `isNotEmpty()`

zromano replied:
wow, nice catch ha fixed

apederson94 commented on src/integration-test/java/SheetResourcesIT.java:
```diff
@@ -179,28 +205,25 @@ public void testCreateSheetInFolder() throws SmartsheetException, IOException {
-        }
+        assertThat(newSheetFromTemplate.getId().toString()).isEmpty();
+        assertThat(newSheetFromTemplate.getAccessLevel()).isEqualTo(AccessLevel.OWNER);
+        assertThat(newSheetFromTemplate.getPermalink()).isEmpty();
```
Should be `isNotEmpty()`

zromano replied:
wow, nice catch ha fixed

zromano replied:
Wonder why the tests didn't fail?

apederson94 commented on src/integration-test/java/SheetResourcesIT.java:
```diff
@@ -210,46 +233,43 @@ public void testCreateSheetInWorkspaceFromTemplate() throws SmartsheetException,
-                || newSheetFromTemplate.getPermalink().toString().isEmpty()) {
-            fail("Sheet not correctly copied");
-        }
+        assertThat(newSheetFromTemplate.getId().toString()).isEmpty();
```
Should be `isNotEmpty()`

zromano replied:
fixed

apederson94 commented on src/integration-test/java/SheetResourcesIT.java:
```diff
@@ -210,46 +233,43 @@ public void testCreateSheetInWorkspaceFromTemplate() throws SmartsheetException,
-        }
+        assertThat(newSheetFromTemplate.getId().toString()).isEmpty();
+        assertThat(newSheetFromTemplate.getAccessLevel()).isEqualTo(AccessLevel.OWNER);
+        assertThat(newSheetFromTemplate.getPermalink()).isEmpty();
```
Should be `isNotEmpty()`

zromano replied:
fixed

apederson94 commented on src/integration-test/java/SheetResourcesIT.java:
```diff
@@ -305,27 +325,27 @@ public void testUpdatePublishStatus() throws SmartsheetException, IOException {
         //PagedResult<Sheet> sheets = smartsheet.sheetResources().listOrganizationSheets(parameters);
 
-        //assertTrue(sheets.getPageNumber() == 1);
+        //assertThat(sheets.getPageNumber() == 1).isTrue();
```
Should we just remove this if it's going to remain commented out?

zromano replied:
see other comment

apederson94 commented on src/integration-test/java/UserResourcesIT.java:
```diff
@@ -90,14 +93,14 @@ public void testAddUser() throws IOException, SmartsheetException {
 //        User user = new User.UpdateUserBuilder().setAdmin(true).setUserId(userId).setFirstName("Adi").setLicensedSheetCreator(true).build();
 //        User updatedUser = smartsheet.userResources().updateUser(user);
-//        assertNotNull(updatedUser);
+//        assertThat(updatedUser).isNotNull();
```
Should we just remove this if it will remain commented out?

zromano replied:
see other comment

apederson94 commented on src/integration-test/java/UserResourcesIT.java:
```diff
@@ -90,14 +93,14 @@ public void testAddUser() throws IOException, SmartsheetException {
         //PagedResult<Sheet> sheets = smartsheet.userResources().listOrgSheets();
         //not executed in test due to low permission
-        //assertNotNull(sheets);
+        //assertThat(sheets).isNotNull();
```
Should we just remove this if it will remain commented out?

zromano replied:
same

apederson94 commented on src/main/java/com/smartsheet/api/CommentAttachmentResources.java:
```diff
@@ -105,5 +105,5 @@ public interface CommentAttachmentResources {
 //     * @throws SmartsheetException if there is any other error during the operation
 //     */
-//    public Attachment attachFileWithMultipartUpload(long sheetId, Comment comment, File file, String contentType) throws FileNotFoundException, SmartsheetException;
+//    Attachment attachFileWithMultipartUpload(long sheetId, Comment comment, File file, String contentType) throws FileNotFoundException, SmartsheetException;
```
Should we just remove this if it will remain commented out?

zromano replied:
same

apederson94 commented on src/main/java/com/smartsheet/api/internal/AbstractResources.java:
```diff
@@ -983,24 +979,16 @@ private static void copyStream(InputStream input, OutputStream output) throws IO
-            } catch (UnsupportedEncodingException e) {
-                throw new RuntimeException ("Unsupported encode. You must support utf-8 for the Smartsheet Java SDK to work",e);
-            }
+            headers.put("Assume-User", URLEncoder.encode(smartsheet.getAssumedUser(), StandardCharsets.UTF_8));
```
I assume that using `StandardCharsets.UTF_8` should never throw, correct?

zromano replied:
Yeah, it won't have the possibility of throwing anymore since we are using the enum and not a string

apederson94 commented on src/main/java/com/smartsheet/api/internal/RowColumnResourcesImpl.java:
```diff
@@ -266,11 +267,7 @@ private void addImage(String path, InputStream inputStream, String contentType,
-        } catch (UnsupportedEncodingException e) {
-            throw new RuntimeException(e);
-        }
+        request.getHeaders().put("Content-Disposition", "attachment; filename=\"" + URLEncoder.encode(imageName, StandardCharsets.UTF_8) + "\"");
```
Can we separate the string addition here onto a separate line? It's a bit messy as is

zromano replied:
done.

apederson94 commented on src/main/java/com/smartsheet/api/internal/SearchResourcesImpl.java:
```diff
@@ -136,17 +143,12 @@ public SearchResult search(String query, EnumSet<SearchInclusion> includes, Sear
-        } catch (UnsupportedEncodingException e) {
-            throw new RuntimeException(e);
-        }
+        return this.getResource("search/sheets/" + sheetId + "?query=" + URLEncoder.encode(query,
```
Can we separate the string concatenation to another line? This is a tad messy

zromano replied:
done.

apederson94 commented on src/main/java/com/smartsheet/api/internal/SheetSummaryResourcesImpl.java:
```diff
@@ -322,18 +339,14 @@ private Result<SummaryField> addSheetSummaryFieldImage(String path, InputStream
-        } catch (UnsupportedEncodingException e) {
-            throw new RuntimeException(e);
-        }
+        request.getHeaders().put("Content-Disposition", "attachment; filename=\"" + URLEncoder.encode(imageName, StandardCharsets.UTF_8) + "\"");
```
Can we move the string concatenation onto a separate line with a variable assignment maybe? This is a bit messy

zromano replied:
done

apederson94 commented on src/main/java/com/smartsheet/api/internal/SheetUpdateRequestResourcesImpl.java:
```diff
@@ -194,7 +193,7 @@ public PagedResult<SentUpdateRequest> listSentUpdateRequests(long sheetId, Pagin
      */
     public SentUpdateRequest getSentUpdateRequest(long sheetId, long sentUpdateRequestId) throws SmartsheetException {
-        return this.getResource("sheets/" + sheetId + "/sentupdaterequests/" + sentUpdateRequestId, 
+        return this.getResource("sheets/" + sheetId + "/sentupdaterequests/" + sentUpdateRequestId,
```
Can we move the string concat to a separate variable?

zromano replied:
done for all the ones in this file

apederson94 commented on src/main/java/com/smartsheet/api/internal/UserResourcesImpl.java:
```diff
@@ -415,15 +435,11 @@ private User attachProfileImage(String path, String file, String contentType) th
-        } catch (UnsupportedEncodingException e) {
-            throw new RuntimeException(e);
-        }
+        request.getHeaders().put("Content-Disposition", "attachment; filename=\"" + URLEncoder.encode(file, StandardCharsets.UTF_8) + "\"");
```
Can we move the string concat to a separate variable?

zromano replied:
done.

apederson94 commented on src/main/java/com/smartsheet/api/internal/http/DefaultHttpClient.java:
```diff
@@ -219,7 +225,7 @@ public HttpResponse request(HttpRequest smartsheetRequest) throws HttpClientExce
                     requestEntityCopy = new HttpEntitySnapshot(entity);
                 } catch (IOException iox) {
-                    logger.error("failed to make copy of original request entity - {}", iox);
+                    logger.error("failed to make copy of original request entity", iox);
```
Should we retain the dash and space, or will the logger add some of that anyways?

zromano replied:
the logger will format it properly

apederson94 commented on src/main/java/com/smartsheet/api/internal/http/DefaultHttpClient.java:
```diff
@@ -496,7 +502,7 @@ public void releaseConnection() {
                 apacheHttpResponse.close();
                 apacheHttpResponse = null;
             } catch (IOException e) {
-                logger.error("error closing Apache HttpResponse - {}", e);
```
Should we retain the dash and space, or will the logger add some of that anyways?

zromano replied:
the logger will format it properly

apederson94 commented on src/main/java/com/smartsheet/api/internal/oauth/OAuthFlowImpl.java:
```diff
@@ -291,8 +305,6 @@ public Token obtainNewToken(AuthorizationResult authorizationResult) throws  OAu
      *
      * @param token the token to refresh
      * @return the refreshed token
-     * @throws NoSuchAlgorithmException the no such algorithm exception
```
Should this one have been removed?

zromano replied:
Yeah, it doesn't actually throw that exception so it's results in a javadoc error

apederson94 commented on src/main/java/com/smartsheet/api/internal/util/QueryUtil.java:
```diff
@@ -20,34 +20,33 @@
     }
 
-    public static String generateUrl(String baseUrl, Map<String, Object> parameters) {
+    public static String generateUrl(@Nullable String  baseUrl, @Nullable Map<String, ? extends Object> parameters) {
```
Looks like `baseUrl` has 2 spaces between it and it's type

zromano replied:
wow nice catch. Fixed

apederson94 commented on src/main/java/com/smartsheet/api/internal/util/QueryUtil.java:
```diff
@@ -60,21 +59,17 @@ public static String generateUrl(String baseUrl, Map<String, Object> parameters)
+        for(Map.Entry<String, ? extends Object> entry : parameters.entrySet()) {
+            // Check to see if the key/value isn't null or empty string
+            if (entry.getKey() != null && (entry.getValue() != null && !entry.getValue().toString().equals(""))) {
+                result.append('&').append(URLEncoder.encode(entry.getKey(), StandardCharsets.UTF_8)).append("=")
+                        .append(URLEncoder.encode(entry.getValue().toString(), StandardCharsets.UTF_8));
```
Can we move the `URLEncoder.encode()` code out to separate variables to enhance readability?

zromano replied:
done.

apederson94 commented on src/test/java/com/smartsheet/api/internal/GroupResourcesImplTest.java:
```diff
@@ -56,36 +55,36 @@ void testGetGroups() throws SmartsheetException, IOException {
+        assertThat(groups.getData().get(0).getCreatedAt()).isNotNull();
+        assertThat(groups.getData().get(0).getModifiedAt()).isNotNull();
+        assertThat(groups.getData().get(0).getDescription()).isNotNull();
+        //assertThat(groups.getData().get(1).getId()).isNotNull();
```
Should we remove this if it is to remain commented out?

zromano replied:
same comment. Will approach in another MR

apederson94 commented on src/test/java/com/smartsheet/api/internal/json/ObjectValueDeserializerTest.java:
```diff
@@ -268,38 +270,38 @@ void longObjectValue() throws IOException, JSONSerializerException {
         ObjectValue actual = getObjectValue(json);
         PrimitiveObjectValue<Number> numberPrimitiveObjectValue = (PrimitiveObjectValue<Number>) actual;
-        assertEquals(123456.789f, numberPrimitiveObjectValue.getValue().floatValue(), 0.001);
+        assertThat(numberPrimitiveObjectValue.getValue().floatValue()).isEqualTo(123456.789f);
```
It looks like this isn't quite right. We are missing the delta here of 0.001 that was in the `assertEquals` implementation

zromano replied:
fixed

apederson94 commented on src/test/java/com/smartsheet/api/internal/util/QueryUtilTest.java:
```diff
@@ -20,83 +20,150 @@
-        String commaSeparatedStringList = QueryUtil.generateCommaSeparatedList(list);
-        assertEquals("123456789,987654321", commaSeparatedStringList);
+    @Nested
+    class GenerateCommaSeparatedList {
```
These nested tests are a nice improvement!

zromano replied:
Thank you 👍

zromano replied:
I think they make the tests easier to read and easier to move around if a method moves classes

apederson94 commented on src/test/java/com/smartsheet/api/logging/LoggingIT.java:
```diff
@@ -0,0 +1,95 @@
+        client.setTraces(Trace.Request, Trace.Response);    // should log entire request and response
+
+        // Note this requires an internet connection
+        assertThatThrownBy(() ->client.sheetResources().getSheet(42, null, null, null, null, null, 1, 1))
```
We are missing a space between the `->` and `client`

zromano replied:
Fixed. One of my next improvements will be to add checkstyle ha

apederson94 commented on src/test/java/com/smartsheet/api/logging/LoggingIT.java:
```diff
@@ -0,0 +1,95 @@
+    void testCustomLogging() {
+        ByteArrayOutputStream traceStream = new ByteArrayOutputStream();
+        DefaultHttpClient.setTraceStream(traceStream);
+        Smartsheet client = new SmartsheetBuilder().setAccessToken("ll352u9jujauoqz4gstvsae05").build(); // using "null" as token results in NPE
```
Might make sense to extract this token to a variable called `dummyToken` or something like that

zromano replied:
made it clear it's a dummy token

apederson94 commented on src/test/java/com/smartsheet/api/models/SheetEmailFormatTest.java:
```diff
@@ -21,19 +21,19 @@
-        assertEquals(3,SheetEmailFormat.values().length);
+        assertThat(SheetEmailFormat.valueOf("PDF")).isNotNull();
+        assertThat(SheetEmailFormat.valueOf("EXCEL")).isNotNull();
+        assertThat(SheetEmailFormat.valueOf("PDF_GANTT")).isNotNull();
```
It doesn't look like this was in the original testing logic. Is this a valid use case to test?

zromano replied:
Yeah, it was just missing a test for one of the enum values.

apederson94 commented on src/test/java/com/smartsheet/api/sdk_test/HelperFunctions.java:
```diff
@@ -36,7 +37,9 @@ public static Smartsheet SetupClient(String apiScenario){
 	}
-	public static void ExceptionMessage(String message, Throwable cause){
+
+	// Todo: fix this... there is no point of this logging
```
We should probably document this in an issue or somewhere with more visibility so it doesn't get lost and then never fixed

zromano replied:
I will open an issue and fix this in a subsequent MR 👍

zromano replied:
I actually just went ahead an fixed this. I updated all the `sdk_tests` to no longer use a strange try-catch format

apederson94 commented on src/test/java/com/smartsheet/api/sdk_test/RowTest.java:
```diff
@@ -205,9 +205,9 @@ void addRows_AssignValues_Bool()
-			assertEquals(true, addedRows.get(0).getCells().get(0).getValue());
-			assertEquals(101L, addedRows.get(0).getCells().get(0).getColumnId().longValue());
+			assertThat(addedRows.get(0).getId().intValue()).isEqualTo(10);
+			assertThat(addedRows.get(0).getCells().get(0).getValue()).isEqualTo(true);
```
Should we use `isTrue()` here instead?

zromano replied:
we can't here because it doesn't recognize/know that it's a boolean type

apederson94 commented on src/test/java/com/smartsheet/api/sdk_test/RowTest.java:
```diff
@@ -530,8 +530,8 @@ void UpdateRows_AssignValues_Bool()
-			assertEquals(101L, updatedRows.get(0).getCells().get(0).getColumnId().longValue());
-			assertEquals(true, updatedRows.get(0).getCells().get(0).getValue());
+			assertThat(updatedRows.get(0).getCells().get(0).getColumnId().longValue()).isEqualTo(101L);
+			assertThat(updatedRows.get(0).getCells().get(0).getValue()).isEqualTo(true);
```
Should we use `isTrue()` here instead?

zromano replied:
we can't here because it doesn't recognize/know that it's a boolean type

apederson94 commented on src/test/java/com/smartsheet/api/sdk_test/RowTest.java:
```diff
@@ -765,8 +765,8 @@ void UpdateRows_ClearValue_Checkbox()
-			assertEquals(101L, updatedRows.get(0).getCells().get(0).getColumnId().longValue());
-			assertEquals(false, updatedRows.get(0).getCells().get(0).getValue());
+			assertThat(updatedRows.get(0).getCells().get(0).getColumnId().longValue()).isEqualTo(101L);
+			assertThat(updatedRows.get(0).getCells().get(0).getValue()).isEqualTo(false);
```
Should we use `isFalse()` here?

zromano replied:
we can't here because it doesn't recognize/know that it's a boolean type

apederson94 reviewed (dismissed):
I left some comments that ask about additional cleanup work which might be out of scope, but since we're already cleaning up a bunch of stuff here we may as well do that as well. Overall some really great work you've done here! Thanks!

smar-zach-romano committed:
Separate string concatenation into separate variables and update invalid test assertions

smar-zach-romano committed:
Update SDK tests so that they do not have a try-catch format
- If an exception is thrown, just let the test fail. No need to catch it and fail the test manually

apederson94 commented on src/test/java/com/smartsheet/api/internal/json/ObjectValueDeserializerTest.java:
```diff
@@ -38,6 +38,7 @@
 
 class ObjectValueDeserializerTest {
     private static final double DELTA = 0.000001;
+    private static final float DELTA_FLOAT = 0.000001f;
```
This value was previously `0.001f`. It should probably stay the same unless you have a valid reason to change it

smar-zach-romano committed:
Fix float value assertion error tolerance
